api/internal/target/kusttarget.go

```diff
@@ -412,11 +412,16 @@ func (kt *KustTarget) accumulateResources(
 			}
 			ldr, err := kt.ldr.New(path)
 			if err != nil {
-				if kusterr.IsMalformedYAMLError(errF) { // Some error occurred while tyring to decode YAML file
-					return nil, errF
+				if strings.Contains(err.Error(), load.ErrRtNotDir.Error()) { // Was neither a remote resource nor a local directory.
```

Comment:
I looked through the code paths again with fresh eyes, and although I agree that the regression here would be with the error message, I think there are several cases where it would happen. Essentially, we know at this point that the target was not a valid individual _file_ (local or remote), and that it is also not a valid _base_ (local or remote), but it may still be a valid _directory_ or an _invalid_ file. Examples of cases where this PR currently changes the error message and should not:
- Path is blank (will incorrectly say it was accumulating a remote resource)
- Path is absolute (will incorrectly say it was accumulating a remote resource)
- Path is a remote file containing invalid YAML (the yaml error will now be swallowed) <-- this is the worst one, try `https://github.com/kubernetes-sigs/kustomize/blob/master/ARCHITECTURE.md?submodules=false` to see it for example
- Path creates a reference cycle (will say, sometimes incorrectly, it was accumulating a remote resource)

This code is complicated, also under-covered, and generates bad error messages for a lot of cases, including the one you pointed out here. I would love to see it improve, but in light of the above I would like to remove changes to it from this PR so we can get this merged.